Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -770,6 +773,10 @@ void RenderGrid::placeItemsOnGrid(GridTrackSizingAlgorithm& algorithm, std::opti
     }
 #endif
 
+    auto masonryAxisDirection = style().gridMasonryRows() ? ForRows : ForColumns;
+    auto gridAxisDirection = masonryAxisDirection == ForRows ? ForColumns : ForRows;
+    unsigned gridAxisTracksBeforeAutoPlacement = grid.numTracks(gridAxisDirection);
```

Comment:
Should be fixed now let me know what you think. I think there's still some room for improvement, but it might be fine as the implementation gets worked out.